beacon-chain/core/state/transition.go

```diff
@@ -100,7 +106,211 @@ func ProcessBlock(state *pb.BeaconState, block *pb.BeaconBlock) (*pb.BeaconState
 
 // NewEpochTransition describes the per epoch operations that are performed on the
 // beacon state.
-func NewEpochTransition(state *pb.BeaconState) *pb.BeaconState {
-	// TODO(#1074): This will encompass all the related logic to epoch transitions.
-	return state
+//
+// Spec pseudocode definition:
+// 	 process_candidate_receipt_roots(state)
+// 	 update_justification(state)
+// 	 update_finalization(state)
+// 	 update_crosslinks(state)
+// 	 process_casper_reward_penalties(state)
+// 	 process_crosslink_reward_penalties(state)
+// 	 update_validator_registry(state)
+// 	 final_book_keeping(state)
+func NewEpochTransition(state *pb.BeaconState) (*pb.BeaconState, error) {
+
+	// Calculate total balances of active validators of the current state.
```

Comment:
https://github.com/ethereum/eth2.0-specs/blob/master/specs/core/0_beacon-chain.md#helpers
from L121 to L178